src/lib/auth/authconf.rb

```diff
@@ -0,0 +1,1150 @@
+# encoding: utf-8
+# ------------------------------------------------------------------------------
+# Copyright (c) 2016 SUSE LINUX GmbH, Nuernberg, Germany.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact SUSE Linux GmbH.
+#
+# ------------------------------------------------------------------------------
+#
+# Module: Manipulate system authentication configuration.
+# Author: Howard Guo <hguo@suse.com>
+
+require 'yast'
+require 'resolv'
+require 'open3'
+require 'socket'
+require 'tempfile'
+require 'fileutils'
+require 'date'
+
+module Auth
+    class AuthConf
+        include Yast
+        include I18n
+        include Logger
+        include UIShortcuts
+
+        # Return network host name and environment facts.
+        def self.get_net_facts
+            resolvable_name = ''
+            begin
+                resolvable_name = Socket.gethostbyname(Socket.gethostname)
+                resolvable_name = resolvable_name.fetch(0, '')
+            rescue
+                # Intentionally ignored
+            end
+            ip_addresses = []
+            begin
+                resolvable_ip = Socket.getaddrinfo(Socket.gethostname, nil)
+                resolvable_ip = resolvable_ip.fetch(0, []).fetch(3, '')
+                if resolvable_ip != ''
+                    ip_addresses = [resolvable_ip]
+                end
+            rescue
+                # Just get interface IPs
+                ip_addresses = Socket.getifaddrs.select{|iface| iface.addr.ip?}.map{|iface| iface.addr.ip_address}.select{|addr| !addr.start_with?('127.')}
+            end
+            domain_name, status = Open3.capture2('domainname')
```

Comment:
in general I recommend to use standard yast ways like SCR.Execute (old way) or new one with Yast::Execute - https://github.com/yast/yast-yast2/blob/6514ef601d7fcb4555c1f8d30a447cb3bc29209a/library/system/src/lib/yast2/execute.rb